libbpf-rs/src/map.rs

```diff
@@ -409,16 +426,81 @@ impl Map {
     pub fn keys(&self) -> MapKeyIter {
         MapKeyIter::new(self, self.key_size())
     }
+
+    /// Create the bpf map standalone.
+    pub fn create_map(
+        name: &str,
+        map_type: MapType,
+        key_size: u32,
+        value_size: u32,
+        max_entries: u32,
+        flags: MapCreateFlags,
+    ) -> Result<Map> {
+        let opts = libbpf_sys::bpf_map_create_opts {
+            sz: std::mem::size_of::<libbpf_sys::bpf_map_create_opts>() as libbpf_sys::size_t,
+            map_flags: flags.bits as i32,
+            btf_fd: 0,
+            btf_key_type_id: 0,
+            btf_value_type_id: 0,
+            btf_vmlinux_value_type_id: 0,
+            inner_map_fd: 0,
+            map_extra: 0,
+            numa_node: 0,
+            map_ifindex: 0,
+        };
+
+        let map_name_c = util::str_to_cstring(name)?;
+        let fd = unsafe {
+            libbpf_sys::bpf_map_create(
+                map_type.clone() as u32,
```

Comment:
I will return `Result<Map>`:

```rust
        Ok(Map {
            fd,
            name: name.to_string(),
            ty: map_type as u32,
            key_size,
            value_size,
            ptr: std::ptr::null_mut(),
        })
```
 
Because the `MapType` is not copyable and the `map_type` need to use in `bpf_map_create()`,  then I use the `clone()` to reuse the variable.

Reply:
I think let MapType derive `num_enum::IntoPrimitive` is a better choice.

Reply:
I'm not familiar with `num_enum` crate, I will try your advice after reading the docs.